segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/PravegaConnectionListenerTest.java

```diff
@@ -195,4 +200,21 @@ public void testCreateRequestProcessor() {
                 mock(StreamSegmentStore.class), mock(TableStore.class), NoOpScheduledExecutor.get());
         Assert.assertTrue(listener.createRequestProcessor(new TrackedConnection(new ServerConnectionInboundHandler())) instanceof AppendProcessor);
     }
+
+    // Test the health status created with pravega listener.
+    @Test
+    public void testHealth() {
+        HealthServiceManager healthServiceManager = new HealthServiceManager(Duration.ofSeconds(2));
+        healthServiceManager.start();
+        int port = TestUtils.getAvailableListenPort();
+        @Cleanup
+        PravegaConnectionListener listener = new PravegaConnectionListener(false, false, "localhost",
+                port, mock(StreamSegmentStore.class), mock(TableStore.class), SegmentStatsRecorder.noOp(),
+                TableSegmentStatsRecorder.noOp(), new PassingTokenVerifier(), null, null, true,
+                NoOpScheduledExecutor.get(), TLS_PROTOCOL_VERSION.getDefaultValue().split(","), healthServiceManager);
+
+        listener.startListening();
+        Health health = listener.getHealthServiceManager().getHealthSnapshot();
+        Assert.assertEquals("HealthContributor should report an 'UP' Status.", Status.UP, health.getStatus());
```

Comment:
Check other cases, like when you close the listener this should report `DOWN`, right? Please, think carefully what are the cases you can find here.